core/shared/src/main/scala/com/monovore/decline/opts.scala

```diff
@@ -133,6 +134,33 @@ object Opts {
     Single(Opt.Regular(namesFor(long, short), metavarFor[A](metavar), help, visibility))
       .mapValidated(Argument[A].read)
 
+  def optionalOptArg[A](
+      long: String,
+      help: String,
+      short: String = "",
+      metavar: String = "",
+      visibility: Visibility = Visibility.Normal
+  )(implicit arg: Argument[A]): Opts[Option[A]] =
+    Single(Opt.OptionalOptArg(namesFor(long, short), metavarFor[A](metavar), help, visibility))
+      .mapValidated {
+        case None => Validated.Valid(None)
+        case Some(value) => arg.read(value).map(Some(_))
+      }
+
+  def optionalOptArgs[A](
+      long: String,
+      help: String,
+      short: String = "",
+      metavar: String = "",
+      visibility: Visibility = Visibility.Normal
+  )(implicit arg: Argument[A]): Opts[(Int, List[A])] =
```

Comment:
What do you think of `NonEmptyList[Option[A]]` instead? (Preserves more ordering information, and it's a "tighter" type in the sense that users don't have to stress about invalid values like `(0, Nil)`.)